src/AutoEncoderModel.cpp

```diff
@@ -124,6 +124,15 @@ AutoEncoderModel::AutoEncoderModel() :
     else {
         printf("No %s found. Initializing new flow decoder model.\n", flowDecoderFilename); // TODO logging
     }
+
+    // Create output directories if they do not exist
+    const std::string modelsRoot{"models"};
+    if (not fs::exists(modelsRoot)) {
+        printf("Default models directory does not exist. Trying to create the directory\n");
+        if (not fs::create_directory(fs::path(modelsRoot))) {
+            printf("Could not create the directory for models. Expect a crash upon training\n");
+        }
+    }
```

Comment:
Is it really the best idea to have the model directory creation logic in one of the models? Wouldn't `Trainer` be the better option for example? So we don't need to duplicate all this into the next model we implement? (I'd assume that we'll keep reusing the same directory)